examples/temperature-measurement-app/esp32/main/include/diagnostic-logs-provider-delegate-impl.h

```diff
@@ -39,7 +45,16 @@ namespace DiagnosticLogs {
 class LogProvider : public DiagnosticLogsProviderDelegate
 {
 public:
+#ifdef CONFIG_ENABLE_ESP_DIAGNOSTICS_TRACE
+    static inline LogProvider & GetInstance(CircularDiagnosticBuffer * bufferInstance)
```

Comment:
Either there should be a setter API for setting bufferInstance or may be a constructor override?

Reply:
we should document the behaviour in case of `nullptr`.